src/ProjectTemplates/Web.ProjectTemplates/content/EmptyWeb-CSharp/Program.cs

```diff
@@ -1,26 +1,14 @@
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
-using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Configuration;
+using Microsoft.AspNetCore.Builder;
 using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
 
-namespace Company.WebApplication1
-{
-    public class Program
-    {
-        public static void Main(string[] args)
-        {
-            CreateHostBuilder(args).Build().Run();
-        }
+await using var app = WebApplication.Create(args);
 
-        public static IHostBuilder CreateHostBuilder(string[] args) =>
-            Host.CreateDefaultBuilder(args)
-                .ConfigureWebHostDefaults(webBuilder =>
-                {
-                    webBuilder.UseStartup<Startup>();
-                });
-    }
+if (app.Environment.IsDevelopment())
+{
+    app.UseDeveloperExceptionPage();
 }
+
+app.MapGet("/", (Func<string>)(() => "Hello World!"));
```

Comment:
I don't like that we still have to explicitly cast to `Func<string>` until we can consume the changes in https://github.com/dotnet/roslyn/pull/52448, but I still think this is already a big improvement overall.

Reply:
@jaredpar @cston

Reply:
The bits should be lining up for preview 4 to allow this cast to be removed.